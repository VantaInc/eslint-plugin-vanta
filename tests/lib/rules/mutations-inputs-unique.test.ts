import { GraphQLRuleTester } from "@graphql-eslint/eslint-plugin";
import rule from "../../../lib/rules/mutations-inputs-unique";

const ruleTester = new GraphQLRuleTester();

ruleTester.runGraphQLTests("mutations-inputs-unique", rule, {
  valid: [
    {
      code: `
type Mutation {
  validMutation(input: validInput!): abcPayload
  anotherOne(input: anotherInput!): bcdPayload
}
`,
    },
    {
      code: `
type Mutation {
  validMutation(input: validInput!): abcPayload
}

extend type Mutation {
  anotherOne(input: anotherInput!): bcdPayload
}
`,
    },
    {
      code: `
type Mutation {
    validMutation(input: validInput!): abcPayload
    anotherOne(input: anotherInput!): bcdPayload
    thirdOne: cdePayload
}
`,
    },
    {
      code: `
  type Mutation {
      validMutation: abcPayload
      anotherOne: bcdPayload
  }
  `,
    },
  ],
  invalid: [
    {
      code: `
type Mutation {
  firstMutation(input: Input!): abcPayload
  secondMutation(input: Input!): bcdPayload
}
`,
      errors: 1,
    },
    {
      code: `
type Mutation {
  firstMutation(input: abcInput!): abcPayload
  secondMutation(input: bcdInput!): bcdPayload
  thirdMutation(input: abcInput!): cdePayload
}
`,
      errors: 1,
    },
    {
      code: `
    type Mutation {
      firstMutation(input: Input!): abcPayload
    }

    extend type Mutation {
      secondMutation(input: Input!): cdePayload}
    `,
      errors: 1,
    },
    {
      code: `
    type Mutation {
      firstMutation(input: abcInput!): abcPayload
      secondMutation(input: bcdInput!): bcdPayload
      thirdMutation(input: abcInput!): cdePayload
    }

    extend type Mutation {
      fourthMutation(input: bcdInput!): defPayload
    }
    `,
      errors: 2,
    },
  ],
});
