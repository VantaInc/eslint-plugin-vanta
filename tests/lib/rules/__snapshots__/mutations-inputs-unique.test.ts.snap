// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invalid #1 1`] = `
"#### ⌨️ Code

      1 | type Mutation {
      2 |   firstMutation(input: Input!): abcPayload
      3 |   secondMutation(input: Input!): bcdPayload
      4 | }

#### ❌ Error

    > 1 | type Mutation {
        | ^^^^^^^^^^^^^^^
    > 2 |   firstMutation(input: Input!): abcPayload
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 3 |   secondMutation(input: Input!): bcdPayload
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 4 | }
        | ^^ Mutation input types must be unique; reused type Input"
`;

exports[`Invalid #2 1`] = `
"#### ⌨️ Code

      1 | type Mutation {
      2 |   firstMutation(input: abcInput!): abcPayload
      3 |   secondMutation(input: bcdInput!): bcdPayload
      4 |   thirdMutation(input: abcInput!): cdePayload
      5 | }

#### ❌ Error

    > 1 | type Mutation {
        | ^^^^^^^^^^^^^^^
    > 2 |   firstMutation(input: abcInput!): abcPayload
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 3 |   secondMutation(input: bcdInput!): bcdPayload
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 4 |   thirdMutation(input: abcInput!): cdePayload
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 5 | }
        | ^^ Mutation input types must be unique; reused type abcInput"
`;

exports[`Invalid #3 1`] = `
"#### ⌨️ Code

      1 |     type Mutation {
      2 |       firstMutation(input: Input!): abcPayload
      3 |     }
      4 |
      5 |     extend type Mutation {
      6 |       secondMutation(input: Input!): cdePayload}

#### ❌ Error

    > 1 |     type Mutation {
        | ^^^^^^^^^^^^^^^^^^^
    > 2 |       firstMutation(input: Input!): abcPayload
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 3 |     }
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 4 |
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 5 |     extend type Mutation {
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 6 |       secondMutation(input: Input!): cdePayload}
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Mutation input types must be unique; reused type Input"
`;

exports[`Invalid #4 1`] = `
"#### ⌨️ Code

      1 |     type Mutation {
      2 |       firstMutation(input: abcInput!): abcPayload
      3 |       secondMutation(input: bcdInput!): bcdPayload
      4 |       thirdMutation(input: abcInput!): cdePayload
      5 |     }
      6 |
      7 |     extend type Mutation {
      8 |       fourthMutation(input: bcdInput!): defPayload
      9 |     }

#### ❌ Error 1/2

    > 1 |     type Mutation {
        | ^^^^^^^^^^^^^^^^^^^
    > 2 |       firstMutation(input: abcInput!): abcPayload
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 3 |       secondMutation(input: bcdInput!): bcdPayload
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 4 |       thirdMutation(input: abcInput!): cdePayload
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 5 |     }
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 6 |
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 7 |     extend type Mutation {
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 8 |       fourthMutation(input: bcdInput!): defPayload
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 9 |     }
        | ^^^^^^ Mutation input types must be unique; reused type abcInput

#### ❌ Error 2/2

    > 1 |     type Mutation {
        | ^^^^^^^^^^^^^^^^^^^
    > 2 |       firstMutation(input: abcInput!): abcPayload
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 3 |       secondMutation(input: bcdInput!): bcdPayload
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 4 |       thirdMutation(input: abcInput!): cdePayload
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 5 |     }
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 6 |
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 7 |     extend type Mutation {
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 8 |       fourthMutation(input: bcdInput!): defPayload
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 9 |     }
        | ^^^^^^ Mutation input types must be unique; reused type bcdInput"
`;
